rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -1532,14 +1533,42 @@ private MessageCall getMessageCall(DataWord gas, DataWord codeAddress, Activatio
         // the callee will receive less gas than the parent expected.
         long calleeGas = Math.min(remainingGas, specifiedGasPlusMin);
 
+        // TODO:I next two lines can be skipped for RSKIP validation. Temp vars so RSKIP pseudocode var names match with the used ones. Later `remainingGas` and `gasCost` should be reused
+        long availableGas = remainingGas;
+        long standardCallCosts = gasCost;
+
         if (computeGas) {
-            gasCost = GasCost.add(gasCost, calleeGas);
-            spendOpCodeGas();
+            if (activations.isActive(RSKIP209)) {
+                program.spendGas(standardCallCosts, op.name());
+                long callGas = Math.min(specifiedGasPlusMin, availableGas); // TO_CHECK_1 @sergio to take into account transfer stipend, we cannot use `top-of-stack item` but `specifiedGasPlusMin`
+                long lockGas = availableGas / 64;
+                long maxGasToPassCallee = availableGas - lockGas; // TO_CHECK_2 @sergio we are not taking into account transfer stipend here
+                if (callGas > maxGasToPassCallee) {
+                    Map<Integer, Long> lockedGasByDepth = this.program.getLockedGasByDepth();
+                    long prevGasLock = lockedGasByDepth.getOrDefault(program.getCallDeep(), 0L);
+                    long forbiddenGas = callGas - maxGasToPassCallee;
+                    if (forbiddenGas > prevGasLock) { // locked for depth can only increase
+                        long expectedLock = forbiddenGas - prevGasLock;
+                        program.spendGas(expectedLock, "Locked for CALL depth " + program.getCallDeep());
+                        program.futureRefundGas(expectedLock); // TO_CHECK_4 @sergio this is how we were handling future refunds
+                        lockedGasByDepth.put(program.getCallDeep(), forbiddenGas);
+                        availableGas -= expectedLock; // TO_CHECK_5 @sergio this line is required to properly detect `movedRemainingGasToChild`, check usage below
+                    }
+                }
+                // TODO:I next two lines can be skipped for RSKIP validation. Temp vars so RSKIP pseudocode var names match with the used ones. Later `calleeGas` and `availableGas` should be reused
+                calleeGas = maxGasToPassCallee;
+                remainingGas = availableGas;
+            } else {
+                // TO_CHECK_6 @sergio aren't we missing consuming `maxGasToPassCallee` after RSKIP209? following lines show previous behavior:
+                // we were increasing `gasCost` with `maxGasToPassCallee` and consuming it inside `spendOpCodeGas()`, now we are just expending `gasCost`+`expectedLock`
+                gasCost = GasCost.add(gasCost, calleeGas);
+                spendOpCodeGas();
```

Comment:
This two lines should be executed for both paths. RSKIP209 and not RSKIP209.

Why only if RSKIP209 is not activated?

Reply:
Yes, that was my doubt, as back then I was not sure while checking the RSKIP pseudocode.

But now I have more context on how this piece of code works (thanks to other task), so I will do another iteration on this algorithm and then we can take another look.